### ISSUE TITLE
Various new undefined-variable false-positives

### ISSUE THREAD

**Issue by The-Compiler** [PyCQA/pylint#1044]
When running the current pylint/astroid master over qutebrowser I get:

```
************* Module qutebrowser.config.style
E: 33, 1: Undefined variable 'functools' (undefined-variable)
E: 44,31: Undefined variable 'template_str' (undefined-variable)
E: 45,11: Undefined variable 'template' (undefined-variable)
E: 45,33: Undefined variable 'colordict' (undefined-variable)

************* Module qutebrowser.config.config
E:655,15: Undefined variable 'self' (undefined-variable)
E:660,19: Undefined variable 'self' (undefined-variable)
E:660,33: Undefined variable 'sectname' (undefined-variable)
E:662,15: Undefined variable 'fallback' (undefined-variable)
E:663,23: Undefined variable 'fallback' (undefined-variable)
E:664,43: Undefined variable 'sectname' (undefined-variable)
E:666,18: Undefined variable 'sect' (undefined-variable)
E:666,23: Undefined variable 'optname' (undefined-variable)
E:668,15: Undefined variable 'fallback' (undefined-variable)
E:669,23: Undefined variable 'fallback' (undefined-variable)
E:670,42: Undefined variable 'optname' (undefined-variable)
E:670,51: Undefined variable 'sectname' (undefined-variable)
E:671,11: Undefined variable 'raw' (undefined-variable)
E:672,19: Undefined variable 'val' (undefined-variable)
E:673,52: Undefined variable 'sect' (undefined-variable)
E:674,17: Undefined variable 'self' (undefined-variable)
E:674,48: Undefined variable 'self' (undefined-variable)
E:674,54: Undefined variable 'sectname' (undefined-variable)
E:674,64: Undefined variable 'optname' (undefined-variable)
E:675,48: Undefined variable 'val' (undefined-variable)
E:675,61: Undefined variable 'mapping' (undefined-variable)
E:676,11: Undefined variable 'transformed' (undefined-variable)
E:677,21: Undefined variable 'val' (undefined-variable)
E:677,39: Undefined variable 'newval' (undefined-variable)
E:678,15: Undefined variable 'newval' (undefined-variable)
```

Those are all false - some are imports, others are `self`. Not sure what's going on there.

**Comment by The-Compiler** [comment]
For a more minimal example:

```python
import functools

@functools.lru_cache()
def foo():
    pass
```

```
E:  3, 1: Undefined variable 'functools' (undefined-variable)
W:  1, 0: Unused import functools (unused-import)
```

**Comment by PCManticore** [comment]
Wow, neither do I. I don't remember touching anything related to undefined-variable for this release. This is kind of a blocker tbh.

**Comment by PCManticore** [comment]
Probably related to https://github.com/PyCQA/astroid/commit/6fba2eca13450d4949bc2925b3a0f4aa3157d968. If I disable that plugin, then this error goes away, so some bad interaction is going on between it and pylint.

**Comment by PCManticore** [comment]
At least this fixes the problem with functools, will have to look for the others as well

**Comment by The-Compiler** [comment]
Indeed, commenting the `MANAGER.register_transform` there gets rid of all those `undefined-variable` warnings, as well as those (which I got for normal `self._foo` access):

```
************* Module qutebrowser.config.config
W:655,15: Access to a protected member _initialized of a client class (protected-access)
W:674,17: Access to a protected member _interpolation of a client class (protected-access)
```

**Comment by PCManticore** [comment]
This should be fixed in the astroid's master. Can you try it, please and then close the issue if it works?

**Comment by The-Compiler** [comment]
Looks good now - thanks!
